Allow "eval $EDITOR" with no arguments.

watcol opened the pull request:
I added edita.vim to Vim and wrote `.bashrc` as following:
```bash
alias vim="eval $EDITOR"
```
And executed `vim` with no arguments, this error is occurred:
```error
Error detected while processing command line..function edita#vim#client#open:
line    2:
E684: list index out of range: -1
E116: Invalid arguments for function fnamemodify(argv()[-1], ':p')
```

Then I fixed `autoload/edita/*/client.vim`, It worked, but new problem was arisen: Client dosen't quit vim when the new buffer was closed. So I patched `autoload/edita/*/editor.vim` to work it.

Now it works very well, please merge this patch.

Note that my environment:
> - Linux 5.11.15 (x86_64)
> - Vim 8.2.2803 (with `+terminal`)
> - Not tested in Neovim

lambdalisue reviewed (changes requested):
Sorry for the late. I haven't noticed this PR.

Well, please rebase all changes and make commits a bit more proper so that I can review them.

watcol committed:
feat: Allow "eval $EDITOR" with no arguments

Use empty string instead of a target path, when no arguments
are provided in "eval $EDITOR".

watcol committed:
fix: Enables s:BufDelete() to delete empty buffer.

To get a buffer variable, use buffer numbers instead of
file names.

watcol commented:
I refactored the commit message, sorry for my careless.

lambdalisue commented on autoload/edita/neovim/client.vim:
```diff
@@ -4,7 +4,8 @@ function! edita#neovim#client#open() abort
   let ch = sockconnect(mode, server, { 'rpc': 1 })
-  let target = fnamemodify(argv()[-1], ':p')
+  let args = argv()
+  let target = len(args) != 0 ? fnamemodify(args[-1], ':p') : ""
```
Please use `isnot#` instead of `!=`

lambdalisue reviewed (changes requested):
One more small fix please 🙏

watcol committed:
fix: Use "isnot#" instead of "!="

lambdalisue commented:
Thanks a lot
